### PR TITLE
[Slack] Fix connector deletion when OAuth isn't valid.

### PR DESCRIPTION
**Pull request by lasryaric**

### CODE REVIEW

File: connectors/src/connectors/slack/index.ts [hunk 1]
```diff
@@ -235,16 +235,23 @@ export async function cleanupSlackConnector(
...
+            new Error(
+              `Could not uninstall the Slack app from the user's workspace. Error: ${deleteRes.error}`
+            )
+          );
+        }
+      } catch (e) {
+        logger.error(
+          { connectorId: connectorId, error: e },
+          "Could not uninstall the Slack app from the user's workspace."
         );
```
Comment (spolu) [inline connectors/src/connectors/slack/index.ts:255]: Should we return new Err as well?

Comment (lasryaric) [reply connectors/src/connectors/slack/index.ts:255]: No, the goal of this fix was to be able to delete a connector even when the access token is revoked. So we log it for auditing, but we keep going.

Comment (spolu) [reply connectors/src/connectors/slack/index.ts:255]: But then why do you return a new Err just above?

Comment (spolu) [reply connectors/src/connectors/slack/index.ts:255]: As per IRL we should error here. This silent error will create more problems than it solve. It's better to have an explicit error to the user (us) than silently fail and create API errors on each webhook

**Pull request merged**